plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -844,4 +866,19 @@ public interface TransportSupplier
         TTransport createTransport()
                 throws TTransportException;
     }
+
+    private static String prependCatalogToDbName(
+            @Nullable String catalogName, @Nullable String dbName)
+    {
+        StringBuilder buf = new StringBuilder().append('@').append(catalogName).append("#");
+        if (dbName != null) {
+            if (dbName.isEmpty()) {
+                buf.append("!");
```

Comment:
can you pls point out to hive code / docs where this convention is being employed?

Reply:
from org.apache.hadoop.hive.metastore.utils.MetastoreUtils.prependNotNullCatToDbName

Reply:
please add a code comment as @findinpath requested